### PR TITLE
Add range support

### PR DESCRIPTION
**Pull request by sudo-panda**
This PR aims to add Range function to func_adl_xAOD.

As an example you can run `test_select_hitpattern_of_global_muons` test to check it out.

### CODE REVIEW

File: func_adl_xAOD/cms/aod/event_collections.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import func_adl_xAOD.common.cpp_types as ctyp
+
```
Comment (gordonwatts) [inline func_adl_xAOD/cms/aod/event_collections.py:2]: Run the command "Organize Imports" in vscode to sort and make pretty import statements.

File: func_adl_xAOD/common/ast_to_cpp_translator.py [hunk 2]
```diff
@@ -581,6 +581,7 @@ def visit_Call_Member(self, call_node):
         # method name we are going to be calling against.
         calling_against = self.get_rep(call_node.func.value)
         function_name = call_node.func.attr
+        args = call_node.args
```
Comment (gordonwatts) [inline func_adl_xAOD/common/ast_to_cpp_translator.py:584]: Perhaps define this closer to where it gets used?

File: func_adl_xAOD/common/cpp_types.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 class terminal:
     'Represents something we cannot see inside, like float, or int, or bool'
 
-    def __init__(self, t: str, is_pointer: bool = False):
+    def __init__(self, t: str, is_pointer: bool = False, is_reference: bool = False):
```
Comment (gordonwatts) [inline func_adl_xAOD/common/cpp_types.py:7]: If this isn't needed, it should be removed.

File: func_adl_xAOD/common/statement.py [hunk 4]
```diff
@@ -140,7 +140,7 @@ def emit(self, e):
 
 
 class set_var:
-    'Set a variable to a value'
+    'Set a value to a variable'
```
Comment (gordonwatts) [inline func_adl_xAOD/common/statement.py:143]: Set a variable to a value is correct, but perhaps a better way to write this is "assign a value to a variable".

File: func_adl_xAOD/common/statement.py [hunk 5]
```diff
@@ -153,6 +153,22 @@ def emit(self, e):
         e.add_line('{0} = {1};'.format(self._target.as_cpp(), self._value.as_cpp()))
 
 
+class decl_var_w_ctor:
```
Comment (gordonwatts) [inline func_adl_xAOD/common/statement.py:156]: We need to understand if we really want a second way to declare variables. Could the ctor functionality be implemented elsewhere?

**Review by gordonwatts — CHANGES_REQUESTED** [review]
This is great! Thanks for this!!!

File: func_adl_xAOD/cms/aod/event_collections.py (hunk 1 above)
Comment (sudo-panda) [reply func_adl_xAOD/cms/aod/event_collections.py:2]: Which all files does this needs to be run in?

**Comment by gordonwatts** [comment]
Are we ready to merge this? Should this no longer be hanging out here?

File: func_adl_xAOD/cms/aod/event_collections.py (hunk 1 above)
Comment (sudo-panda) [reply func_adl_xAOD/cms/aod/event_collections.py:2]: > Which all files does this needs to be run in?
@gordonwatts Could you please answer this

**Review by gordonwatts — APPROVED** [review]

**Pull request merged**